src/main/resources/jupyter/init-actions.sh

```diff
@@ -302,7 +303,9 @@ if [[ "${ROLE}" == 'Master' ]]; then
         JUPYTER_USER_SCRIPT=`basename ${JUPYTER_USER_SCRIPT_URI}`
         docker cp /etc/${JUPYTER_USER_SCRIPT} ${JUPYTER_SERVER_NAME}:${JUPYTER_HOME}/${JUPYTER_USER_SCRIPT}
         retry 3 docker exec -u root ${JUPYTER_SERVER_NAME} chmod +x ${JUPYTER_HOME}/${JUPYTER_USER_SCRIPT}
-        retry 3 docker exec -u root -e PIP_USER=false ${JUPYTER_SERVER_NAME} ${JUPYTER_HOME}/${JUPYTER_USER_SCRIPT}
+        docker exec -u root -e PIP_USER=false ${JUPYTER_SERVER_NAME} ${JUPYTER_HOME}/${JUPYTER_USER_SCRIPT} &> us_output.txt || EXIT_CODE=$? && true ;
+        gsutil cp us_output.txt ${JUPYTER_USER_SCRIPT_OUTPUT_URI}
+        $EXIT_CODE
```

Comment:
Not sure if this does anything..

Reply:
I tested this by running Terra locally while pointing to a fiab of this branch. For a cluster with a failing userscript, the init script stops at this line, whereas if the script is successful it passes this line. I think because we're using `set -e` at the beginning, this will work, but I can try to see if `return $EXIT_CODE` also works if that's more readable

Reply:
I just tried using return and it looks like it doesn't work, going to stick with the original!

Reply:
Yeah, I think this works because of `set -e` but can I suggest a change for readability?
```
if [ $EXIT_CODE -ne 0 ]; then
    echo "User script failed with exit code $EXIT_CODE. Output is saved to $JUPYTER_USER_SCRIPT_OUTPUT_URI."
    exit $EXIT_CODE
fi
```
This way we're not relying on `set -e` semantics and it's more clear IMO.

Reply:
Sounds good!